src/node/nodestate.h

```diff
@@ -763,14 +763,13 @@ namespace ccf
 
       if (network.consensus_type == ConsensusType::PBFT)
       {
-        // for now do not encrypt the ledger as the current implementation does
-        // not work for PBFT
-        recovery_encryptor = std::make_shared<NullTxEncryptor>();
+        recovery_encryptor =
+          std::make_shared<PbftTxEncryptor>(network.ledger_secrets, true);
       }
       else
```

Comment:
can you please put else if RAFT here and then have an else case where we have some kind of error

Reply:
Done